TestFileIcns::test_save_append_images test fails (different content)

### What did you do?
Built pillow-5.1.0 (from Fink distribution)

### What did you expect to happen?
All tests should pass, like they do for 5.0.0 (though 5.0.0 doesn't have TestFileIcns::test_save_append_images)

### What actually happened?
TestFileIcns::test_save_append_images fails

### What versions of Pillow and Python are you using?
pillow-5.1.0 against Fink-python-2.7.15, -3.4.8, -3.5.5, and -3.6.5. All 4 python version fail this test.

````python
/sw/bin/pytest-2.7 -vx Tests || exit 2
============================================================ test session starts =============================================================
platform darwin -- Python 2.7.15, pytest-3.5.1, py-1.5.3, pluggy-0.6.0 -- /sw/bin/python2.7
cachedir: .pytest_cache
rootdir: /sw/build.build/pil-py27-5.1.0-1/Pillow-5.1.0, inifile: setup.cfg
plugins: hypothesis-3.34.0
collected 1123 items                                                                                                                         
....
Tests/test_file_icns.py::TestFileIcns::test_save_append_images FAILED                                                                  [ 15%]

================================================================== FAILURES ==================================================================
____________________________________________________ TestFileIcns.test_save_append_images ____________________________________________________

self = <test_file_icns.TestFileIcns testMethod=test_save_append_images>

    @unittest.skipIf(sys.platform != 'darwin', "requires macOS")
    def test_save_append_images(self):
        im = Image.open(TEST_FILE)
    
        temp_file = self.tempfile("temp.icns")
        provided_im = Image.new('RGBA', (32, 32), (255, 0, 0, 0))
        im.save(temp_file, append_images=[provided_im])
    
        reread = Image.open(temp_file)
>       self.assert_image_equal(reread, im)

Tests/test_file_icns.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
Tests/helper.py:115: in assert_image_equal
    self.fail(msg or "got different content")
E   AssertionError: got different content
============================================== 1 failed, 176 passed, 2 skipped in 7.65 seconds ===============================================

--- a/Tests/test_file_icns.py
+++ b/Tests/test_file_icns.py
@@ -40,11 +40,11 @@ class TestFileIcns(PillowTestCase):
         im = Image.open(TEST_FILE)
 
         temp_file = self.tempfile("temp.icns")
-        provided_im = Image.new('RGBA', (32, 32), (255, 0, 0, 0))
+        provided_im = Image.new('RGBA', (32, 32), (255, 0, 0, 128))
         im.save(temp_file, append_images=[provided_im])
 
         reread = Image.open(temp_file)
-        self.assert_image_equal(reread, im)
+        self.assert_image_similar(reread, im, 1)
 
         reread = Image.open(temp_file)
         reread.size = (16, 16, 2)

